module/x/gravity/keeper/ethereum_event_handler.go

```diff
@@ -34,79 +34,36 @@ func (a EthereumEventProcessor) Handle(ctx sdk.Context, eve types.EthereumEvent)
 				return sdkerrors.Wrapf(err, "mint vouchers coins: %s", coins)
 			}
 		}
+
 		return a.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, addr, coins)
+
 	case *types.BatchExecutedEvent:
 		a.keeper.batchTxExecuted(ctx, common.HexToAddress(event.TokenContract), event.BatchNonce)
 		return
-	case *types.ERC20DeployedEvent:
-		// Check if it already exists
-		if existingERC20, exists := a.keeper.getCosmosOriginatedERC20(ctx, event.CosmosDenom); exists {
-			return sdkerrors.Wrap(
-				types.ErrInvalid,
-				fmt.Sprintf("ERC20 %s already exists for denom %s", existingERC20.Hex(), event.CosmosDenom))
-		}
-
-		// Check if denom exists
-		// TODO: document that peggy chains require denom metadata set
-		metadata := a.keeper.bankKeeper.GetDenomMetaData(ctx, event.CosmosDenom)
-		if metadata.Base == "" {
-			return sdkerrors.Wrap(types.ErrInvalid, fmt.Sprintf("denom not found %s", event.CosmosDenom))
-		}
-
-		// Check if attributes of ERC20 match Cosmos denom
-		if event.Erc20Name != metadata.Display {
-			return sdkerrors.Wrap(
-				types.ErrInvalid,
-				fmt.Sprintf("ERC20 name %s does not match denom display %s", event.Erc20Name, metadata.Description))
-		}
 
-		if event.Erc20Symbol != metadata.Display {
-			return sdkerrors.Wrap(
-				types.ErrInvalid,
-				fmt.Sprintf("ERC20 symbol %s does not match denom display %s", event.Erc20Symbol, metadata.Display))
-		}
-
-		// ERC20 tokens use a very simple mechanism to tell you where to display the decimal point.
-		// The "decimals" field simply tells you how many decimal places there will be.
-		// Cosmos denoms have a system that is much more full featured, with enterprise-ready token denominations.
-		// There is a DenomUnits array that tells you what the name of each denomination of the
-		// token is.
-		// To correlate this with an ERC20 "decimals" field, we have to search through the DenomUnits array
-		// to find the DenomUnit which matches up to the main token "display" value. Then we take the
-		// "exponent" from this DenomUnit.
-		// If the correct DenomUnit is not found, it will default to 0. This will result in there being no decimal places
-		// in the token's ERC20 on Ethereum. So, for example, if this happened with Atom, 1 Atom would appear on Ethereum
-		// as 1 million Atoms, having 6 extra places before the decimal point.
-		// This will only happen with a Denom Metadata which is for all intents and purposes invalid, but I am not sure
-		// this is checked for at any other point.
-		decimals := uint32(0)
-		for _, denomUnit := range metadata.DenomUnits {
-			if denomUnit.Denom == metadata.Display {
-				decimals = denomUnit.Exponent
-				break
-			}
-		}
-
-		if decimals != uint32(event.Erc20Decimals) {
-			return sdkerrors.Wrap(
-				types.ErrInvalid,
-				fmt.Sprintf("ERC20 decimals %d does not match denom decimals %d", event.Erc20Decimals, decimals))
+	case *types.ERC20DeployedEvent:
+		if err := a.verifyERC20DeployedEvent(ctx, event); err != nil {
```

Comment:
It looks like you took out the part that selects between the two ERC20 metadata options depending on whether the metadata is set?

Reply:
NVM I see that I was wrong